mautic/mautic
# Add support for plugin to inject buttons into other bundle pages.

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | n
| New feature? | y
| Related user documentation PR URL | na
| Related developer documentation PR URL | todo
| Issues addressed (#s or URLs) | na
| BC breaks? | n
| Deprecations? | n

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
This PR adds support for plugins and other bundles to inject buttons into the various standard button areas of Mautic without requiring cross hard coding.

It also moves the batch buttons to be an activated dropdown around the check all. 

![2016-11-19_14-25-45](https://cloud.githubusercontent.com/assets/63312/20466143/38a9a5fa-af22-11e6-92d7-7d8524054210.gif)

#### Steps to test this PR:
1. Extract the attached zip and drop attached ButtonSubscriber.php and config.php into the LeadBundle to test
2. Clear cache
3. View /s/contacts and /s/contacts/view/ID pages. On the /s/contacts, note the second "Extra Button" action for each contact dropdown. Also note the star/extra button in the batch toolbar area (on every page with the attached example file).
4. Go to a /s/contacts/view/ID page and note the Extra Button in the page actions/button list.
5. Ensure that the page actions have the correct formatting for the primary/buttons and dropdown list along with the dropdowns in the lists. Toolbar area should still include some buttons on the contacts page. 
6. For /s/contacts (and other list views), checking at least one items should activate the batch dropdown list. Unchecking them all should disable it.

[example.zip](https://github.com/mautic/mautic/files/602323/example.zip)
